multichain-aggregator/multichain-aggregator-logic/src/services/search.rs

```diff
@@ -394,6 +396,39 @@ pub async fn search_domains(
     Ok((domains, next_page_token))
 }
 
+pub async fn search_domains_cached(
+    cache: Option<&DomainSearchCache>,
+    bens_client: HttpApiClient,
+    query: String,
+    protocols: Option<Vec<String>>,
+    primary_chain_id: ChainId,
+    page_size: u32,
+    page_token: Option<String>,
+) -> Result<(Vec<Domain>, Option<String>), ServiceError> {
+    let key = format!(
+        "{}:{}:{}:{}:{}",
+        query,
+        protocols.as_ref().map(|p| p.join(",")).unwrap_or_default(),
+        primary_chain_id,
+        page_size,
```

Comment:
As I understand, cache for search with `page_size=5` (and empty token) won't help for exactly the same search but `page_size=1` (quick search??). Is it true?

If so, I'm thinking about ways to improve this; restricting page size is not quite good

Maybe we can at least populate `page_size`=1 cache additionally when querying larger sizes? Vice versa won't work but it sounds useful IMO

Reply:
>Is it true?

Yes

Currently, BENS domain lookups do not support full text search. It searches only for full input string matches. So, the result set for a particular chain id is always either 0 or 1 element.

As a simple optimization I decided to set a default page_size for all queries to 50 (`api_settings.default_page_size`). This should improve number of cache hits while providing the correct data as before.